RIOT-OS/RIOT
# boards/arduino*: Added/refactored params for W5100

### Contribution description
- ATmega based boards: Added parameter for W5100 to `board_common.h`
- Arduino Due based boards: Use `board.h` to override `W5100_PARAM_CS` and `W5100_PARAM_EVT` instead of providing an adapted copy of `w5100_params.h`

### Testing procedure
Running `USEMODULE=w5100 BOARD=arduino-mega2560 make` in `examples/gnrc_minimal` should successfully build and when flashed should respond to `ping6`. (The example will print the IPv6 address over UART).

The same should still work for `arduino-due`.

### Issues/PRs references
None.
